index.js

```diff
@@ -1,302 +1,293 @@
+	}
+
+	/* eslint-disable new-cap */
+
+	static SignIn(obj) {
+		return this.processMethod('SignIn', obj);
+	}
+
+	static SimpleResponse(obj) {
+		return this.processMethod('SimpleResponse', obj);
+	}
+
+	static NewSurface(obj) {
+		return this.processMethod('NewSurface', obj);
+	}
+
+	static Image(obj) {
+		return this.processMethod('Image', obj);
+	}
+
+	static List(obj) {
+		return this.processMethod('List', obj);
+	}
+
+	static BasicCard(obj) {
+		return this.processMethod('BasicCard', obj);
+	}
+
+	static Button(obj) {
+		return this.processMethod('Button', obj);
+	}
+
+	static isNotEmpty(str) {
+		return str && str.trim() !== '';
+	}
+
+	static complete(convo, action = 'close', options = { log: false, logFunc: null }) {
+		if (!convo.conv){
+			if (options.log) {
+				console.log(`${action}:${JSON.stringify(convo, null, 1)}`);
+			}
+			return Promise.resolve(convo);
+		}
+		let finalText = convo._write.reduce((acc, text) => acc.sentence(text), new Say()).toString().trim();
+		let finalSpeech = convo._speak.reduce((acc, text) => acc.sentence(text), new Say()).toString().trim();
+		let promises = [];
+		let textWasPopulated = !!(finalText || finalSpeech);
+
+		let isRunningInDialogFlow = typeof convo.conv[action] === 'function';
+		let convoFunc = isRunningInDialogFlow ? obj => convo.conv[action](obj) : resp => {
+			if (options.log) {
+				if (options.logFunc) {
+					options.logFunc(action, resp);
+				}
+				else {
+					console.log(`${action}:${JSON.stringify(resp, null, 1)}`);
+				}
+			}
+		};
+
+		if (textWasPopulated) {
+			promises.push(convoFunc(Convo.SimpleResponse({
+				text: finalText || finalSpeech,
+				speech: finalSpeech || finalText
+			})));
+		}
+
+		convo._rich.forEach((richResponse) => {
+			if (richResponse.media && !richResponse.capabilities && !richResponse.send) {
+				promises.push(convoFunc(richResponse.media));
+				return;
+			}
+			if (convo.conv.surface && convo.conv.surface.capabilities.has(richResponse.capabilities)) {
+				if (!textWasPopulated && richResponse.send && (typeof richResponse.media !== 'string')) {
+					promises.push(convoFunc(Convo.SimpleResponse({
+						text: richResponse.send.notification,
+						speech: richResponse.send.notification
+					})));
+				}
+				promises.push(convoFunc(richResponse.media));
+			}
+			else if (richResponse.send && convo.conv.available && convo.conv.available.surfaces.capabilities.has(richResponse.capabilities)) {
+				promises.push(convoFunc(Convo.NewSurface(
+					{
+						context: richResponse.send.context,
+						notification: richResponse.send.notification,
+						capabilities: richResponse.capabilities
+					}
+				)));
+			}
+		});
+
+		return isRunningInDialogFlow ? Promise.all(promises) : Promise.all(promises).then(() => convo);
+	}
+
+	/* eslint-enable new-cap */
+	static prepComplete(obj, action = 'close', options) {
+		if (isPromise(obj)) {
+			return obj.then(result => this.prepComplete(result, action, options));
```

Comment:
I renamed `obj` to `result` in the arrow function to avoid any ambiguities on which scoped `obj` was being used